lib/tasks/gettext.rake

```diff
@@ -31,10 +32,13 @@ begin
       def files_to_translate
         Dir.glob("#{@engine.root}/{app,db,lib,config,locale}/**/*.{rb,erb,haml,slim,rhtml,js}")
       end
+
+      def text_domain
+        @domain
+      end
     end
 
-    Foreman::Gettext::Support.add_text_domain args[:engine], "#{@engine_root}/locale"
-    ENV['TEXTDOMAIN'] = args[:engine]
+    Foreman::Gettext::Support.add_text_domain @domain, "#{@engine_root}/locale"
 
     Rake::Task['gettext:find'].invoke
```

Comment:
For the record, I rarely see that this line fails with Nil exception, the task is sometimes not available:

```
# rake plugin:gettext[foreman_discovery]
...
rake aborted!
NoMethodError: undefined method `match' for nil:NilClass
/home/lzap/work/foreman/lib/tasks/gettext.rake:39:in `block in <top (required)>'
Tasks: TOP => gettext:po:update => gettext:po:ru:update
(See full trace by running task with --trace)
```

Ori reproduced this as well, but since this was there before this patch, I am merging this and we can attempt to find this later.